### PR TITLE
Misc fixes tests

### PR DESCRIPTION
**Pull request by VRehnberg**
This is mainly two things:

 1. Solves https://github.com/ejnnr/cupbearer/issues/19
 2. And marks test_train_mahalanobis_advex as flaky
 
2 is the questionable one. It adds a new dependency to run test-suite and that dependency has [questionable maintenance](https://github.com/box/flaky/issues/192). Additionally, it would be better to just fix test so that it isn't flaky, but I got a bit annoyed and wanted a quick-fix.

Shouldn't need much review. Either cherry-pick only 1 or merge as is.

### CODE REVIEW

**Comment by VRehnberg** [comment]
There are some remaining warnings in tests, mostly from import handling deprecations internally in lightning when I'm using python3.11 but also when deprecation warning for torch.meshgrid, which made me uncertain if we're using it correctly. See https://github.com/pytorch/pytorch/issues/50276

For square images it shouldn't matter, but I'll have to look at it again to be sure that it works in general.

**Comment by VRehnberg** [comment]
So actually looked at why advex test was flaky and at a first glance it shouldn't be (not strict comparisson)
```python
rob_acc > cfg.success_threshold
```
but `rob_acc` in torchattacks is actually in percent https://github.com/Harry24k/adversarial-attacks-pytorch/blob/master/torchattacks/attack.py#L298 so my fix is to scale these to same unit. Possibly we'd like a test that sees that attack can succeed, but with these (basically) untrained models it seems that there are sometimes attacks can't succeed so anything less than 100 % `success_threshold` could fail then.

**Comment by VRehnberg** [comment]
Finally no warnings &#58378;
```
==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.11.5, pytest-7.4.3, pluggy-1.3.0
rootdir: /home/vikren/Documents/Private/SERI-MATS/Research/cupbearer
configfile: pytest.ini
collected 35 items                                                                                                                                                                           

tests/test_abstractions.py ............                                                                                                                                                [ 34%]
tests/test_data.py .............                                                                                                                                                       [ 71%]
tests/test_pipeline.py .......                                                                                                                                                         [ 91%]
tests/test_utils.py ...                                                                                                                                                                [100%]

==================================================================================== 35 passed in 10.59s =====================================================================================
```

File: src/cupbearer/detectors/finetuning.py [hunk 1]
```diff
@@ -102,13 +113,15 @@ class FinetuningTrainConfig(TrainConfig):
     num_epochs: int = 10
     batch_size: int = 128
     max_steps: Optional[int] = None
+    log_every_n_steps: Optional[int] = (None,)
```
Comment (ejnnr) [inline src/cupbearer/detectors/finetuning.py:116]: Is the tuple here a typo?

File: src/cupbearer/scripts/conf/train_classifier_conf.py [hunk 2]
```diff
@@ -25,6 +25,7 @@ class Config(ScriptConfig):
     dir: DirConfig = mutable_field(
         DirConfig, base=os.path.join("logs", "train_classifier")
     )
+    log_every_n_steps: Optional[int] = (None,)
```
Comment (ejnnr) [inline src/cupbearer/scripts/conf/train_classifier_conf.py:28]: Tuple here as well

File: src/cupbearer/scripts/make_adversarial_examples.py [hunk 3]
```diff
@@ -35,7 +35,8 @@ def main(cfg: Config):
     )
     rob_acc, l2, elapsed_time = atk.save(dataloader, save_path, return_verbose=True)
 
-    if rob_acc > cfg.success_threshold:
+    # N.B. rob_acc is in percent while success_threshold is not
+    if rob_acc > 100 * cfg.success_threshold:
         raise RuntimeError(
             f"Attack failed, new accuracy is {rob_acc} > {cfg.success_threshold}."
```
Comment (ejnnr) [inline src/cupbearer/scripts/make_adversarial_examples.py:41]:
```suggestion
            f"Attack failed, new accuracy is {rob_acc}% > {100 * cfg.success_threshold}%."
```

**Review by ejnnr — COMMENTED** [review]
Thanks, looks great! Only thing I noticed is two places where you have a tuple `(None,)` as the default for `log_every_n_steps`. I assume that's a typo but wasn't sure enough to just change it and merge. (If it's deliberate and I'm missing something, then at least the type annotation should be different). Feel free to merge after dealing with that!

File: src/cupbearer/detectors/finetuning.py (hunk 1 above)
Comment (VRehnberg) [reply src/cupbearer/detectors/finetuning.py:116]: Yes, must have had a trailing comma.

**Pull request merged**